editor/src/clj/editor/workspace.clj

```diff
@@ -364,11 +360,14 @@ ordinary paths."
            :header (format "The editor plugin '%s' is not compatible with this version of the editor. Please edit your project dependencies to refer to a suitable version." (resource/proj-path resource))}))
       false)))
 
+(defn- runduce!
+  "Like run!, but with added transducer"
+  [xform f coll]
+  (transduce xform (completing #(f %2)) nil coll)
+  nil)
+
 (defn- load-editor-plugins! [workspace added]
-  (let [added-resources (set (map resource/proj-path added))
-        plugin-resources (find-clojure-plugins workspace)
-        plugin-resources (filter (fn [x] (contains? added-resources (resource/proj-path x))) plugin-resources)]
-    (dorun (map (fn [x] (load-plugin! workspace x)) plugin-resources))))
+  (runduce! (filter is-plugin-clojure-file?) #(load-plugin! workspace %) added))
```

Comment:
Improved performance. The previous implementation went through all project resources, filtered them for Clojure files, then filtered once more to only include added files. There was no need to get all files since filtering added files for Clojure is enough.